### ISSUE TITLE
Add training + eval pipeline for probabilistic forecasting

### ISSUE THREAD

**Issue by alyssadsouza** [Blood-Glucose-Control/nocturnal-hypo-gly-prob-forecast#403]
Similar to our workflow script + eval script we need probabilistic aware training + probabilistic aware evaluation metrics.

Specifically for chronos2:
1. All of our configurations use `eval_metric: "RMSE"`, so even though training of chronos2 is actually being trained with WQL, model selection is happening with "RMSE". So first step is updating configs so that they use WQL.
2. eval script is only recording RMSE in the nocturnal_results.jsons... Our script is already computing WQL and BrierAt3.9.  We should wire these in to be recorded in our results so that we can report on them and add them to our analysis.   There should be an early check in the eval pipeline on the supports_probabilistic_forecast() property to see if this eval should continue. 
3. We need a script that generates Probability Integral Transform (PIT) Histograms, I can write the code for the PIT values and the plot but the script should have an obvious place for calling this each episode:
```python
def compute_pit_values(quantile_forecasts, actuals, quantile_levels):
    """Compute PIT values from quantile forecasts.
    
    Args:
        quantile_forecasts: (n_quantiles, forecast_length)
        actuals: (forecast_length,)
        quantile_levels: list of quantile levels
    Returns:
        (forecast_length,) array of PIT values in [0, 1]
    """
```
PIT Shape:
- Flat: Well-calibrated
- U-shaped: Under-dispersed (intervals too narrow)
- Hump-in-middle: Over-dispersed (intervals too wide)
- Left-skewed: Model systematically over-predicts
- Right-skewed: Model systematically under-predicts

4. We need a script that produces Reliability Diagrams, conditional calibration of quantiles. For each quantile q, compute fraction of actual below predict q-quantile; plot vs. identity line.  This means we need the antiques recorded (same as for PIT):
```ep_result["q_forecast"] = q_forecast.tolist()  # (n_quantiles, forecast_length)```

Really the pipeline needs to be reworked in what it is actually recording because our current set up is storing everything in one giant nocturnal_results.json. 
I'm proposing a 3-Tier storage solution for this:

Tier 1: Leaderboard (CSV ~1 row per run)
- We already have what we need with how the summary.csv works.
- Add wql, brier_3.9, mace, coverage_50, coverage_80, sharpness_50, sharpness_80.

Tier 2: Per-episode scalars (Parquet format, 1 row per episode):
- No arrays. One scalar metric per episode, record the anchor and patient_id. 5000 episodes ~ 1MB

Tier 3: Raw forecsats (compressed NumPy .npz, written once, loaded on-demand, rarely needed)
- quantile_forecasts: shape (n_episodes, n_quantiles, forecast_length) e.g., (4243, 21, 96)
- actuals: shape (n_epsisodes, forecast_length)
- episode_ids: string array 
- with compression this shuold be about ~6MB per run. 

Tier 1 gives us best models
Tier 2 gives us per-patient WQL breakdown for distributional results (box-plots to show variance of prediction quality)
Tier 3 allows us to go back and generate PIT histograms, and reliability diagrams. And we can recompute calibration if we decide on new thresholds, we don't need to rerun the model in some contexts where we might need to. 

- We no longer need to store the context_bg, we should be able to pull that easily from our stored date, patient and episode_id. That's a ton of duplicate data storage. 

File Output Layout:

experiments/nocturnal_forecasting/512ctx_96fh/chronos2/2026-03-15_2316_sweep03_25k_resumed_aleppo_2017_finetuned/
    experiment_config.json          # reproducibility metadata (tiny)
    nocturnal_evaluation.log        # text log
    results_summary.json             # Tier 1: overall scalars only
    episodes.parquet                     # Tier 2: per-episode scalar metrics
    forecasts.npz                             # Tier 3: raw arrays (compressed)

-> compute metrics eagerly, store arrays lazily. Compute WQL, Brier, PIT, calibration, coverage, and sharpness during evaluation. Write the scalars to Tier 1 + 2. Write the raw arrays to Tier 3 only for re-analysis.

**Comment by RobotPsychologist** [comment]
@arohig @alyssadsouza I'm adding Arohi to this, because I just realized for this script it also makes sense to add in the shape aware loss functions here and record them in this process.

You can find Soft-DTW in the `pip install tslearn` tslearn.metrics class:

```python
from tslearn.metrics import soft_dtw

def compute_soft_dtw(pred, actual, gamma=1.0):
    """Soft-DTW between predicted and actual BG trajectories.
    
    Args:
        pred: (forecast_length,) point forecast
        actual: (forecast_length,) ground truth
        gamma: smoothing parameter. Lower = closer to hard DTW.
            gamma=1.0 is standard. gamma→0 recovers exact DTW.
    Returns:
        float: Soft-DTW distance (lower = better shape match)
    """
    # tslearn expects (n_timestamps, 1) shape
    return float(soft_dtw(
        pred.reshape(-1, 1),
        actual.reshape(-1, 1),
        gamma=gamma,
    ))
```

AND

`pip install dilate-loss`

import torch
from dilate_loss import dilate_loss

```python
def compute_dilate(pred, actual, alpha=0.5, gamma=0.01):
    """DILATE loss between predicted and actual trajectories.
    
    DILATE = alpha * shape_loss + (1 - alpha) * temporal_loss
    
    Args:
        pred: (forecast_length,) numpy array
        actual: (forecast_length,) numpy array
        alpha: weight between shape (DTW) and timing (TDI). 
            0.5 = equal weight. Higher = more shape emphasis.
        gamma: Soft-DTW smoothing parameter
    Returns:
        tuple: (dilate, shape_loss, temporal_loss)
    """
    # dilate_loss expects (batch, length, channels) tensors
    p = torch.tensor(pred, dtype=torch.float32).reshape(1, -1, 1)
    a = torch.tensor(actual, dtype=torch.float32).reshape(1, -1, 1)
    loss, shape, temporal = dilate_loss(p, a, alpha=alpha, gamma=gamma)
    return float(loss), float(shape), float(temporal)
```

These should both be recorded in Tier 2.
